WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingCategoriesViewModel.kt

```diff
@@ -0,0 +1,223 @@
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+class PrepublishingCategoriesViewModel @Inject constructor(
+    private val getCategoriesUseCase: GetCategoriesUseCase,
+    private val updatePostCategoriesUseCase: UpdatePostCategoriesUseCase,
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) : ScopedViewModel(bgDispatcher) {
+    private var isStarted = false
+    private lateinit var editPostRepository: EditPostRepository
+    private lateinit var siteModel: SiteModel
+    private var updateCategoriesJob: Job? = null
+
+    private val _navigateToHomeScreen = MutableLiveData<Event<Unit>>()
+    val navigateToHomeScreen: LiveData<Event<Unit>> = _navigateToHomeScreen
+
+    private val _navigateToAddCategoryScreen = MutableLiveData<Event<Unit>>()
+    val navigateToAddCategoryScreen: LiveData<Event<Unit>> = _navigateToAddCategoryScreen
+
+    private val _toolbarTitleUiState = MutableLiveData<UiString>()
+    val toolbarTitleUiState: LiveData<UiString> = _toolbarTitleUiState
+
+    private val _snackbarEvents = MediatorLiveData<Event<SnackbarMessageHolder>>()
+    val snackbarEvents: LiveData<Event<SnackbarMessageHolder>> = _snackbarEvents
+
+    private val _uiState: MutableLiveData<UiState> = MutableLiveData()
+    val uiState: LiveData<UiState> = _uiState
+
+    private val _refreshingUiState = MutableLiveData<SwipeToRefreshUiState>()
+    val refreshingUiState: LiveData<SwipeToRefreshUiState> = _refreshingUiState
+
+    fun start(editPostRepository: EditPostRepository, siteModel: SiteModel) {
+        this.editPostRepository = editPostRepository
+        this.siteModel = siteModel
+
+        if (isStarted) return
+        isStarted = true
+
+        init()
+    }
+
+    private fun init() {
+        setToolbarTitleUiState()
+
+        val siteCategories = getSiteCategories()
+        val postCategories = getPostCategories()
+        _uiState.value = ContentUiState(
+                siteCategories = siteCategories,
+                selectedCategoryIds = postCategories.toHashSet()
+        )
+    }
+
+    private fun setToolbarTitleUiState() {
+        _toolbarTitleUiState.postValue(UiStringRes(string.prepublishing_nudges_toolbar_title_categories))
+    }
+
+    fun onBackButtonClicked() {
+        _navigateToHomeScreen.postValue(Event(Unit))
+    }
+
+    fun onAddNewCategoryClicked() {
+        _navigateToAddCategoryScreen.postValue(Event(Unit))
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        updateCategoriesJob?.cancel()
+    }
+
+    fun updateCategories() {
+        if (!hasChanges()) return
+
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
```

Comment:
I'm not sure what the desired behavior is and interested in what your thoughts are? I'm going to check how the setting work, maybe that will provide a hint. The PPN tags updates each keypress (with a delay), so that doesn't happen on back button. I didn't want to do that here because we might overload categories. wdyt?

Reply:
It works the same way in the settings, so I think we are good. Having said that, I think it's not a great UX - we probably should display a dialog and let the user confirm that they are ok loosing their changes. Either way, it's out of scope.

Reply:
> I think it's not a great UX
I'm with you on this ... no notifications success/failure ... I get it, but we are all over the place in the app and we probably won't come to a standard. 🤷